software/contrib/consequencer.md

```diff
@@ -97,16 +106,32 @@ A fixed voltage will select a single pattern and varying voltage (e.g. an envelo
 
 Patterns can be added, removed or updated by updating the relevant list structures at the end of the consequencer.py file in the pattern class (look for the line `class pattern:`). Once patterns are updated make sure you save a copy of the updated file to the EuroPi using yoru favourite method (Thonny / REPL), then restart the Consequencer script.
 
-The syntax should be intuitive. A four-on-the-floor exanmple pattern is shown below. Each `1` or `0` represents a gate or no gate at that point in the sequence.
+The syntax should be intuitive. An example pattern is shown below. Each `1` or `0` represents a gate or no gate at that point in the sequence.
 The mapping of `BD`, `SN`, `HH` is as follows:
 - BD: Output 1
 - SN: Output 2
 - HH: Output 3
 
+Starting with the July 2022 update, steps now also have a probability which is configured using the BdProb, SnProb and HhProb sections as shown below.
+Use a value from `1` to `9` to set the desired level of probability for each step.
+A value of `9` will cause the step to trigger 100% of the time, any value from `1` to `8` will trigger the step from n/9 times.
+If a 100% probability is desired for every step in a pattern,  place a single digit (e.g. `9`) into the BdProb, SnProb or HhProb lists as shown in the second pattern below.
```

Comment:
[optional] This documentation doesn't _exactly_ match the behavior of the code block at around line 55 and it may result in some user confusion. Specifically the doc says 'single digit', but the code will replace the probabilities if the probability string is any length shorter than the step length.

I would suggest making the code and the doc agree.

Reply:
Good catch. Updated.